File: src/2025-03-04-12-20-21--deepseek-vr-0304/javascript/exercises/practice/robot-name/robot-name.js
export class Robot {
  static usedNames = new Set();

  static releaseNames(name) {
    Robot.usedNames.delete(name);
  }

  constructor() {
    this._name = null;
  }

  get name() {
    if (!this._name) {
      do {
        this._name = this._generateName();
      } while (Robot.usedNames.has(this._name));
      Robot.usedNames.add(this._name);
    }
    return this._name;
  }

  set name(value) {
    throw new Error('Cannot modify robot name');
  }

  reset() {
    if (this._name) {
      Robot.releaseNames(this._name);
      this._name = null;
    }
  }

  _generateName() {
    const letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
    const digits = '0123456789';
    let name = '';
    for (let i = 0; i < 2; i++) {
      name += letters[Math.floor(Math.random() * 26)];
    }
    for (let i = 0; i < 3; i++) {
      name += digits[Math.floor(Math.random() * 10)];
    }
    return name;
  }
}

File: src/python-2025-02-27-deepclaude37-rust/python/exercises/practice/robot-name/robot_name.py
import random
import string

class Robot:
    _used_names = set()

    def __init__(self):
        self._name = None
        self._previous_name = None

    @property
    def name(self):
        """Get the robot's name, generating a new one if needed."""
        if self._name is None:
            self._name = self._generate_name()
        return self._name

    def reset(self):
        """Reset the robot to factory settings, wiping its name."""
        if self._name is not None:
            Robot._used_names.remove(self._name)
            self._previous_name = self._name
            self._name = None

    def _generate_name(self):
        """Generate a unique random name in the format of two uppercase letters followed by three digits."""
        while True:
            letters = ''.join(random.choices(string.ascii_uppercase, k=2))
            digits = ''.join(random.choices(string.digits, k=3))
            new_name = letters + digits
            
            if new_name not in Robot._used_names and new_name != self._previous_name:
                Robot._used_names.add(new_name)
                return new_name
